lib/src/pick_map.dart

```diff
@@ -4,6 +4,23 @@ extension NullableMapPick on Pick {
   @Deprecated('Use .asMapOrThrow()')
   Map<RK, RV> Function<RK, RV>() get asMap => asMapOrThrow;
 
+  /// Returns the picked [value] as [Map]
+  ///
+  /// {@template Pick.asMap}
+  /// Provides a view of this map as having [RK] keys and [RV] instances,
+  /// if necessary.
+  ///
+  /// If this map is already a `Map<RK, RV>`, it is returned unchanged.
+  ///
+  /// If this set contains only keys of type [RK] and values of type [RV],
+  /// all read operations will work correctly.
+  /// If any operation exposes a non-[RK] key or non-[RV] value,
+  /// the operation will throw instead.
+  ///
+  /// Entries added to the map must be valid for both a `Map<K, V>` and a
+  /// `Map<RK, RV>`.
+  /// via [Map] cast function
```

Comment:
It's not useful for deep_pick users to explain the generic arguments.
Compare it to the [`asList()` documentation](https://github.com/passsy/deep_pick/blob/master/lib/src/pick_list.dart#L15-L48).

I expect an example and information what would happen for doublicated keys.